### PR TITLE
New arch/multiselecton menu options

### PR DESCRIPTION
**Pull request by jabarros**
## Related Issues
App:

Library PR (if needed):

- [ ] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt [hunk 1]
```diff
@@ -564,6 +711,43 @@ class MainFileListFragment : Fragment(), SortDialogListener, SortOptionsView.Sor
...
+
+    fun getShadowView(): View {
+        return binding.shadowView
+    }
+
+    fun setMessageForEmptyList(message: String) {
+        binding.emptyDataParent.root.visibility = View.VISIBLE
+        binding.emptyDataParent.listEmptyDatasetIcon.visibility = View.GONE
+        binding.emptyDataParent.listEmptyDatasetTitle.visibility = View.GONE
+        binding.emptyDataParent.listEmptyDatasetSubTitle.text = message
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt:740]: we can reduce this using kotlin's sintax: `binding.emptyDataParent.let{}`

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt [hunk 2]
```diff
@@ -564,6 +711,43 @@ class MainFileListFragment : Fragment(), SortDialogListener, SortOptionsView.Sor
...
+    }
+
+    fun setProgressBarAsIndeterminate(indeterminate: Boolean) {
+        Timber.d("Setting progress visibility to %s", indeterminate)
+        binding.shadowView.visibility = View.GONE
+        binding.syncProgressBar.visibility = View.VISIBLE
+        binding.syncProgressBar.isIndeterminate = indeterminate
+        binding.syncProgressBar.postInvalidate()
```
Comment (fesave) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt:748]: same here! 😄

**Review by fesave — CHANGES_REQUESTED** [review]
Great! Some tiny suggestions:

File: owncloudApp/src/main/java/com/owncloud/android/presentation/adapters/filelist/FileListAdapter.kt [hunk 3]
```diff
@@ -243,7 +245,7 @@ class FileListAdapter(
                 holder.itemView.setBackgroundColor(Color.WHITE)
                 checkBoxV.setImageResource(R.drawable.ic_checkbox_blank_outline)
             }
-            checkBoxV.visibility = View.VISIBLE
+            checkBoxV.visibility = if (getCheckedItems().isNotEmpty()) View.VISIBLE else View.GONE
```
[the remark this answers is not in the document]
Comment (jabarros) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/adapters/filelist/FileListAdapter.kt]: 👍

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt (hunk 1 above)
Comment (jabarros) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt]: :+1

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt (hunk 2 above)
Comment (jabarros) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt]: :+1

**Pull request merged**